public/course_lecture.php

```diff
@@ -0,0 +1,211 @@
+<?php require_once(dirname(__FILE__)."/../modules/ensure_logged_in.php"); ?>
+
+<?php
+require_once "../modules/models/user.php";
+require_once "../modules/models/course_section.php";
+require_once "../modules/models/course_section_student.php";
+require_once "../common.php";
+
+$course_page_id = $_GET['id'];
+
+try {
+    $course_lecture = CourseOffering::includes("course")->where(array('id' => $course_page_id));
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+
+try {
+    $course_section_students = CourseSectionStudent::includes(["user", "course_section"])->getAll();
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+try {
+    $marked_entities = MarkedEntity::where(array("course_offering_id" => $course_page_id));
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+try {
+    $course_sections = CourseSection::where(array('offering_id' => $course_page_id));
+
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+try {
+    $teams = Team::where(array('course_offering_id' => $course_page_id));
+
+} catch (PDOException $error) {
+    echo "<br>" . $error->getMessage();
+}
+
+?>
+
+<?php include "templates/header.php"; ?>
+
+<?php if(get_current_role() == 'instructor') { ?>
+    <h2>Student List </h2>
+    <?php
+    $course_students = array();
+
+    foreach ($course_sections as $section) {
+        try {
+            $course_section_student = CourseSectionStudent::includes("user")->where(array('section_id' => $section->id));
+            foreach ($course_section_student as $student){
+                array_push($course_students, $student->user);
```

Comment:
can array_merge be used here instead of looping through $course_section_student?

Reply:
hmmm I don't know what array_merge is

Reply:
ignore this, if you implement my recommendation this won't be necessary

Reply:
tried to make it work with array_merge, but it didn't work